### PR TITLE
libc: expose NEWLIB_LIBC_NANO feature for SDK 0.11.x

### PR DESCRIPTION
**Pull request by galak**
SDK 0.11.x has 2 newlib builds per target similar to the arm embedded
toolchain.  The default when using the toolchain is the "full" newlib
which differs from SDK 0.10.x.  To match behavior we expose
SDK_VERSION and SDK_MAJOR_MINOR to Kconfig and update the expose of
NEWLIB_LIBC_NANO for ZEPHYR_TOOLCHAIN_VARIANT = 'zephyr' and
SDK_MAJOR_MINOR = '0.11'.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

File: lib/libc/Kconfig [hunk 1]
```diff
@@ -40,7 +40,8 @@ endchoice # LIBC_IMPLEMENTATION
 
 if NEWLIB_LIBC
 
-if "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "gnuarmemb"
+if "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "gnuarmemb" || \
+  ("$(ZEPHYR_TOOLCHAIN_VARIANT)" = "zephyr" && "${SDK_MAJOR_MINOR}" = "0.11")
 config NEWLIB_LIBC_NANO
 	bool "Build with newlib-nano C library"
 	default y
```
Comment (nashif) [inline lib/libc/Kconfig]: Why is it needed to enable this by default if it is a tool-chain dependent option? I know this is not part of your change, but I think it is a bad a idea to have this filtering based on SDK being used, what if I have another customer toolchain that supports that?

I think we should remove the conditional and disable this by default and let users enable the Kconfig if their toolchain support newlib-nano and not do that based on toolchain name as defined by zephyr.

**Review by nashif — CHANGES_REQUESTED** [review]

File: cmake/kconfig.cmake [hunk 2]
```diff
@@ -48,6 +48,8 @@ set(ENV{ARCH_DIR}   ${ARCH_DIR})
 set(ENV{DEVICETREE_CONF} ${DEVICETREE_CONF})
```
Comment (SebastianBoe) [inline cmake/kconfig.cmake]: "and update the expose of NEWLIB_LIBC_NANO"

?

File: lib/libc/Kconfig [hunk 3]
```diff
@@ -40,7 +40,8 @@ endchoice # LIBC_IMPLEMENTATION
 
 if NEWLIB_LIBC
 
-if "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "gnuarmemb"
+if "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "gnuarmemb" || \
+  ("$(ZEPHYR_TOOLCHAIN_VARIANT)" = "zephyr" && "${SDK_MAJOR_MINOR}" = "0.11")
```
Comment (SebastianBoe) [inline lib/libc/Kconfig]: Adding this for each SDK version is not going to scale.

Comment (SebastianBoe) [reply lib/libc/Kconfig]: I believe it was explained to me that libc-nano was the only supported newlib variant for zephyr-sdk prior to 0.11.

So shouldn't we omit this version check?

File: lib/libc/Kconfig (hunk 1 above)
Comment (SebastianBoe) [reply lib/libc/Kconfig]: IMHO, toolchains should be expressing what they support with custom Kconfig sources.
Just like SoC's, arch's, and board's are already doing.

So there should exist a HAS_NEWLIB_LIBC_NANO option that is sourced through a toolchain-specific Kconfig file.

Then here we could default to enabling NANO when it is supported.

Disabling NEWLIB_LIBC_NANO by default would increase code size.

**Comment by galak** [comment]
Opened this PR as a different approach to the problem:

https://github.com/zephyrproject-rtos/zephyr/pull/22276

**Comment by galak** [comment]
Closing in favor of #22276

**Pull request closed without merging**